### ISSUE TITLE
Pasting Should Not append Space as a Prefix

### ISSUE THREAD

**Issue by HarryXueTJU** [oliverschwendener/ueli#698]
When trying to paste a string from the clipboard after an inputted string, the ueli will auto append a space as a prefix, this is not friendly to web search, esp. when you use web search to form a url, the %20 will ruin the feature.
e.g. When I copy "test" from somewhere and try to paste it in web search, I input "g?" and paste, I will get "g? test", but I want to have "g?test".

**Comment by oliverschwendener** [comment]
This works fine for me. There is no place were a space will be prepended to the pasted text. I guess that you unintentionally copied the space with the search term. Try pasting to somewhere else and check if the space is there too?

**Comment by HarryXueTJU** [comment]
Hi @oliverschwendener, I restarted my mac and tried again, and I find this issue has gone.

But I really tried to copy once and paste to several places, only ueli append a space there, I don't know why.

Anyway, thanks for your time and reply. If I find this happened again, I will try to record a video of it.

**Comment by oliverschwendener** [comment]
I'll close this issue, feel free to open it up again.

**Comment by HarryXueTJU** [comment]
Hi @oliverschwendener,

It happens again, hope this video can help to describe the status. Thanks

[Screen Recording 2021-04-27 at 4.24.21 PM.zip](https://github.com/oliverschwendener/ueli/files/6383039/Screen.Recording.2021-04-27.at.4.24.21.PM.zip)

**Comment by oliverschwendener** [comment]
You're copying the whitespace from the browser. Try pasting the same value to the browser URL bar for example and you'll see that the whitespace is there too.

**Comment by oliverschwendener** [comment]
See my video:

https://user-images.githubusercontent.com/15727229/116215963-6e016080-a748-11eb-895c-c33849fd5c99.mov

closed

**Comment by HarryXueTJU** [comment]
I can see the space in your video, but from my previous video, you can find that when I paste in the comment input box, I do not get the space as you do.
Also, I tried to paste into the browser's URL bar, and I do not get the space.
For the comment input box, I only get a space only when there is already a word. e.g. When I already have "it" there and I try to paste after "it", I will get "it closed". But I will not get " closed" when I paste into a blank comment input box.
I don't know whether it is a system env issue.

By the way, does it mean we cannot directly use web search for copied strings? Could we add a trim space for the query strings in this feature?

Thanks!

**Comment by oliverschwendener** [comment]
Of course you can copy and paste, there is no additional space added when pasting into ueli. You just have to be careful when copying stuff that there is no additional spaced copied. See here is a video me copying and pasting text:

https://user-images.githubusercontent.com/15727229/116353833-bcba0380-a7f7-11eb-9fe1-f4ea6db8125a.mov

**Comment by HarryXueTJU** [comment]
I do the same thing in my original video, double click a word, copy and paste in ueli and get the space as I described above. I don't know how to be more careful in such kind of operation... It's confusing. Maybe it's the system issue...

**Comment by HarryXueTJU** [comment]
[Screen Recording 2021-04-28 at 7.14.24 PM.zip](https://github.com/oliverschwendener/ueli/files/6391384/Screen.Recording.2021-04-28.at.7.14.24.PM.zip)

**Comment by HarryXueTJU** [comment]
Also, you can find in my screen record above, I copied once and do the same action in ueli and spotlight search, I get a space in ueli but do not get one in spotlight search.

**Comment by oliverschwendener** [comment]
1. I saw your video
2. I replicated your issue in my first video to show you that when you copy a word from the browser by double clicking it, you might copy the leading whitespace as well. Checkout [this article](https://developer.mozilla.org/en-US/docs/Web/API/Document_Object_Model/Whitespace) if you want to know why this could happen.
3. Spotlight removes all leading whitespace when pasting something into it. Try copying something with a leading whitespace and paste it to Spotlight and you'll see. Or check out my screen recoring here where I paste a string with leading whitespace into Spotlight:

https://user-images.githubusercontent.com/15727229/116402875-33bdbf00-a82d-11eb-99c0-75e6f8708c53.mov
